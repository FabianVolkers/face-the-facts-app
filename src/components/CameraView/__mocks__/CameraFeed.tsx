import React from 'react';


class CameraFeed extends React.Component {

    render = jest.fn().mockImplementation(() => {
        return (
            <video
                src='./test_scan_video_large.mp4'
            ></video>
        )
    })
}

export default CameraFeed