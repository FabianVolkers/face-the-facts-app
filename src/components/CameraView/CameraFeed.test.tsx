import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import CameraFeed from './CameraFeed';

let container: HTMLDivElement | null = null;

beforeEach(() => {
	// setup a DOM element as a render target
	container = document.createElement('div');
	document.body.appendChild(container);
});

afterEach(() => {
	// cleanup on exiting
	if (container !== null) {
		unmountComponentAtNode(container);
		container.remove();
		container = null;
	}
});

jest.mock('./CameraFeed');

describe('unittest', () => {
	it('renders correctly', () => {
		// given
		const cameraOpts = {
			x: 0,
			y: 0,
			width: window.innerWidth,
			height: (window.innerWidth * 4) / 3,
			camera: 'rear',
			tapPhoto: true,
			previewDrag: false,
			toBack: true,
			alpha: 1,
		};
		const setCameraReady = jest.fn();

		// when
		render(
			<CameraFeed cameraOpts={cameraOpts} setCameraReady={setCameraReady}></CameraFeed>,
			container
		);

		// then
		expect(container?.getElementsByTagName('video')[0]).toBeInTheDocument();
	});
});
