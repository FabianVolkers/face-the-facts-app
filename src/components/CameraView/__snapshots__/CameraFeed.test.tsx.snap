// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unittest renders correctly 1`] = `
CameraFeed {
  "_reactInternalFiber": FiberNode {
    "_debugHookTypes": null,
    "_debugID": 130,
    "_debugIsCurrentlyTiming": false,
    "_debugNeedsRemount": false,
    "_debugOwner": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 128,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": null,
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": [Circular],
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 1,
      "elementType": [Function],
      "expirationTime": 0,
      "firstEffect": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 132,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": [Circular],
        "_debugSource": Object {
          "columnNumber": 4,
          "fileName": "/Users/fabianvolkers/Developer/face-the-facts/mobile-frontend/face-the-facts/src/components/CameraView/CameraFeed.tsx",
          "lineNumber": 143,
        },
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": null,
        "child": null,
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 128,
        "elementType": "video",
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "autoPlay": true,
          "className": "video-element",
          "id": "camera-video",
          "playsInline": true,
        },
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {
          "autoPlay": true,
          "className": "video-element",
          "id": "camera-video",
          "playsInline": true,
        },
        "ref": Object {
          "current": <video
            autoplay=""
            class="video-element"
            id="camera-video"
            playsinline=""
          />,
        },
        "return": [Circular],
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": <video
          autoplay=""
          class="video-element"
          id="camera-video"
          playsinline=""
        />,
        "tag": 5,
        "treeBaseDuration": 0,
        "type": "video",
        "updateQueue": null,
      },
      "index": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": Object {
        "Component": [Function],
        "context": null,
        "props": Object {
          "cameraOpts": Object {
            "alpha": 1,
            "camera": "rear",
            "height": 1365.3333333333333,
            "previewDrag": false,
            "tapPhoto": true,
            "toBack": true,
            "width": 1024,
            "x": 0,
            "y": 0,
          },
          "setCameraReady": [MockFunction],
        },
        "refProp": null,
        "wrappingComponentProps": null,
      },
      "memoizedState": Object {
        "context": null,
        "mount": true,
        "props": Object {
          "cameraOpts": Object {
            "alpha": 1,
            "camera": "rear",
            "height": 1365.3333333333333,
            "previewDrag": false,
            "tapPhoto": true,
            "toBack": true,
            "width": 1024,
            "x": 0,
            "y": 0,
          },
          "setCameraReady": [MockFunction],
        },
        "wrappingComponentProps": null,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "Component": [Function],
        "context": null,
        "props": Object {
          "cameraOpts": Object {
            "alpha": 1,
            "camera": "rear",
            "height": 1365.3333333333333,
            "previewDrag": false,
            "tapPhoto": true,
            "toBack": true,
            "width": 1024,
            "x": 0,
            "y": 0,
          },
          "setCameraReady": [MockFunction],
        },
        "refProp": null,
        "wrappingComponentProps": null,
      },
      "ref": null,
      "return": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 125,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 125,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": null,
          "_debugSource": null,
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": [Circular],
          "child": null,
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 0,
          "elementType": null,
          "expirationTime": 1073741823,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": null,
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": null,
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": FiberRootNode {
            "callbackExpirationTime": 0,
            "callbackNode": null,
            "callbackPriority": 90,
            "containerInfo": <div>
              <video
                autoplay=""
                class="video-element"
                id="camera-video"
                playsinline=""
              />
            </div>,
            "context": Object {},
            "current": [Circular],
            "finishedExpirationTime": 0,
            "finishedWork": null,
            "firstPendingTime": 0,
            "firstSuspendedTime": 0,
            "hydrate": false,
            "interactionThreadID": 17,
            "lastExpiredTime": 0,
            "lastPingedTime": 0,
            "lastSuspendedTime": 0,
            "memoizedInteractions": Set {},
            "nextKnownPendingLevel": 0,
            "pendingChildren": null,
            "pendingContext": null,
            "pendingInteractionMap": Map {},
            "pingCache": null,
            "tag": 0,
            "timeoutHandle": -1,
          },
          "tag": 3,
          "treeBaseDuration": 0,
          "type": null,
          "updateQueue": Object {
            "baseQueue": Object {
              "callback": null,
              "expirationTime": 1073741823,
              "next": [Circular],
              "payload": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "cameraOpts": Object {
                        "alpha": 1,
                        "camera": "rear",
                        "height": 1365.3333333333333,
                        "previewDrag": false,
                        "tapPhoto": true,
                        "toBack": true,
                        "width": 1024,
                        "x": 0,
                        "y": 0,
                      },
                      "setCameraReady": [MockFunction],
                    }
                  }
                  refProp={null}
                  wrappingComponentProps={null}
                />,
              },
              "priority": 97,
              "suspenseConfig": null,
              "tag": 0,
            },
            "baseState": null,
            "effects": null,
            "shared": Object {
              "pending": null,
            },
          },
        },
        "child": [Circular],
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 0,
        "firstEffect": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 132,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": [Circular],
          "_debugSource": Object {
            "columnNumber": 4,
            "fileName": "/Users/fabianvolkers/Developer/face-the-facts/mobile-frontend/face-the-facts/src/components/CameraView/CameraFeed.tsx",
            "lineNumber": 143,
          },
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": null,
          "child": null,
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 128,
          "elementType": "video",
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "autoPlay": true,
            "className": "video-element",
            "id": "camera-video",
            "playsInline": true,
          },
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "autoPlay": true,
            "className": "video-element",
            "id": "camera-video",
            "playsInline": true,
          },
          "ref": Object {
            "current": <video
              autoplay=""
              class="video-element"
              id="camera-video"
              playsinline=""
            />,
          },
          "return": [Circular],
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": <video
            autoplay=""
            class="video-element"
            id="camera-video"
            playsinline=""
          />,
          "tag": 5,
          "treeBaseDuration": 0,
          "type": "video",
          "updateQueue": null,
        },
        "index": 0,
        "key": null,
        "lastEffect": [Circular],
        "memoizedProps": null,
        "memoizedState": Object {
          "element": <WrapperComponent
            Component={[Function]}
            context={null}
            props={
              Object {
                "cameraOpts": Object {
                  "alpha": 1,
                  "camera": "rear",
                  "height": 1365.3333333333333,
                  "previewDrag": false,
                  "tapPhoto": true,
                  "toBack": true,
                  "width": 1024,
                  "x": 0,
                  "y": 0,
                },
                "setCameraReady": [MockFunction],
              }
            }
            refProp={null}
            wrappingComponentProps={null}
          />,
        },
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": FiberRootNode {
          "callbackExpirationTime": 0,
          "callbackNode": null,
          "callbackPriority": 90,
          "containerInfo": <div>
            <video
              autoplay=""
              class="video-element"
              id="camera-video"
              playsinline=""
            />
          </div>,
          "context": Object {},
          "current": [Circular],
          "finishedExpirationTime": 0,
          "finishedWork": null,
          "firstPendingTime": 0,
          "firstSuspendedTime": 0,
          "hydrate": false,
          "interactionThreadID": 17,
          "lastExpiredTime": 0,
          "lastPingedTime": 0,
          "lastSuspendedTime": 0,
          "memoizedInteractions": Set {},
          "nextKnownPendingLevel": 0,
          "pendingChildren": null,
          "pendingContext": null,
          "pendingInteractionMap": Map {},
          "pingCache": null,
          "tag": 0,
          "timeoutHandle": -1,
        },
        "tag": 3,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": Object {
          "baseQueue": null,
          "baseState": Object {
            "element": <WrapperComponent
              Component={[Function]}
              context={null}
              props={
                Object {
                  "cameraOpts": Object {
                    "alpha": 1,
                    "camera": "rear",
                    "height": 1365.3333333333333,
                    "previewDrag": false,
                    "tapPhoto": true,
                    "toBack": true,
                    "width": 1024,
                    "x": 0,
                    "y": 0,
                  },
                  "setCameraReady": [MockFunction],
                }
              }
              refProp={null}
              wrappingComponentProps={null}
            />,
          },
          "effects": null,
          "shared": Object {
            "pending": null,
          },
        },
      },
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": WrapperComponent {
        "_reactInternalFiber": [Circular],
        "_reactInternalInstance": Object {},
        "context": Object {},
        "props": Object {
          "Component": [Function],
          "context": null,
          "props": Object {
            "cameraOpts": Object {
              "alpha": 1,
              "camera": "rear",
              "height": 1365.3333333333333,
              "previewDrag": false,
              "tapPhoto": true,
              "toBack": true,
              "width": 1024,
              "x": 0,
              "y": 0,
            },
            "setCameraReady": [MockFunction],
          },
          "refProp": null,
          "wrappingComponentProps": null,
        },
        "refs": Object {},
        "state": Object {
          "context": null,
          "mount": true,
          "props": Object {
            "cameraOpts": Object {
              "alpha": 1,
              "camera": "rear",
              "height": 1365.3333333333333,
              "previewDrag": false,
              "tapPhoto": true,
              "toBack": true,
              "width": 1024,
              "x": 0,
              "y": 0,
            },
            "setCameraReady": [MockFunction],
          },
          "wrappingComponentProps": null,
        },
        "updater": Object {
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
        },
      },
      "tag": 1,
      "treeBaseDuration": 0,
      "type": [Function],
      "updateQueue": Object {
        "baseQueue": null,
        "baseState": Object {
          "context": null,
          "mount": true,
          "props": Object {
            "cameraOpts": Object {
              "alpha": 1,
              "camera": "rear",
              "height": 1365.3333333333333,
              "previewDrag": false,
              "tapPhoto": true,
              "toBack": true,
              "width": 1024,
              "x": 0,
              "y": 0,
            },
            "setCameraReady": [MockFunction],
          },
          "wrappingComponentProps": null,
        },
        "effects": null,
        "shared": Object {
          "pending": null,
        },
      },
    },
    "_debugSource": null,
    "actualDuration": 0,
    "actualStartTime": -1,
    "alternate": null,
    "child": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 132,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "columnNumber": 4,
        "fileName": "/Users/fabianvolkers/Developer/face-the-facts/mobile-frontend/face-the-facts/src/components/CameraView/CameraFeed.tsx",
        "lineNumber": 143,
      },
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": null,
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 128,
      "elementType": "video",
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "autoPlay": true,
        "className": "video-element",
        "id": "camera-video",
        "playsInline": true,
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "autoPlay": true,
        "className": "video-element",
        "id": "camera-video",
        "playsInline": true,
      },
      "ref": Object {
        "current": <video
          autoplay=""
          class="video-element"
          id="camera-video"
          playsinline=""
        />,
      },
      "return": [Circular],
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": <video
        autoplay=""
        class="video-element"
        id="camera-video"
        playsinline=""
      />,
      "tag": 5,
      "treeBaseDuration": 0,
      "type": "video",
      "updateQueue": null,
    },
    "childExpirationTime": 0,
    "dependencies": null,
    "effectTag": 5,
    "elementType": [Function],
    "expirationTime": 0,
    "firstEffect": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 132,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "columnNumber": 4,
        "fileName": "/Users/fabianvolkers/Developer/face-the-facts/mobile-frontend/face-the-facts/src/components/CameraView/CameraFeed.tsx",
        "lineNumber": 143,
      },
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": null,
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 128,
      "elementType": "video",
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "autoPlay": true,
        "className": "video-element",
        "id": "camera-video",
        "playsInline": true,
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "autoPlay": true,
        "className": "video-element",
        "id": "camera-video",
        "playsInline": true,
      },
      "ref": Object {
        "current": <video
          autoplay=""
          class="video-element"
          id="camera-video"
          playsinline=""
        />,
      },
      "return": [Circular],
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": <video
        autoplay=""
        class="video-element"
        id="camera-video"
        playsinline=""
      />,
      "tag": 5,
      "treeBaseDuration": 0,
      "type": "video",
      "updateQueue": null,
    },
    "index": 0,
    "key": null,
    "lastEffect": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 132,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "columnNumber": 4,
        "fileName": "/Users/fabianvolkers/Developer/face-the-facts/mobile-frontend/face-the-facts/src/components/CameraView/CameraFeed.tsx",
        "lineNumber": 143,
      },
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": null,
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 128,
      "elementType": "video",
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "autoPlay": true,
        "className": "video-element",
        "id": "camera-video",
        "playsInline": true,
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "autoPlay": true,
        "className": "video-element",
        "id": "camera-video",
        "playsInline": true,
      },
      "ref": Object {
        "current": <video
          autoplay=""
          class="video-element"
          id="camera-video"
          playsinline=""
        />,
      },
      "return": [Circular],
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": <video
        autoplay=""
        class="video-element"
        id="camera-video"
        playsinline=""
      />,
      "tag": 5,
      "treeBaseDuration": 0,
      "type": "video",
      "updateQueue": null,
    },
    "memoizedProps": Object {
      "cameraOpts": Object {
        "alpha": 1,
        "camera": "rear",
        "height": 1365.3333333333333,
        "previewDrag": false,
        "tapPhoto": true,
        "toBack": true,
        "width": 1024,
        "x": 0,
        "y": 0,
      },
      "setCameraReady": [MockFunction],
    },
    "memoizedState": null,
    "mode": 0,
    "nextEffect": null,
    "pendingProps": Object {
      "cameraOpts": Object {
        "alpha": 1,
        "camera": "rear",
        "height": 1365.3333333333333,
        "previewDrag": false,
        "tapPhoto": true,
        "toBack": true,
        "width": 1024,
        "x": 0,
        "y": 0,
      },
      "setCameraReady": [MockFunction],
    },
    "ref": null,
    "return": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 128,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": null,
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": [Circular],
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 1,
      "elementType": [Function],
      "expirationTime": 0,
      "firstEffect": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 132,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": [Circular],
        "_debugSource": Object {
          "columnNumber": 4,
          "fileName": "/Users/fabianvolkers/Developer/face-the-facts/mobile-frontend/face-the-facts/src/components/CameraView/CameraFeed.tsx",
          "lineNumber": 143,
        },
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": null,
        "child": null,
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 128,
        "elementType": "video",
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "autoPlay": true,
          "className": "video-element",
          "id": "camera-video",
          "playsInline": true,
        },
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {
          "autoPlay": true,
          "className": "video-element",
          "id": "camera-video",
          "playsInline": true,
        },
        "ref": Object {
          "current": <video
            autoplay=""
            class="video-element"
            id="camera-video"
            playsinline=""
          />,
        },
        "return": [Circular],
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": <video
          autoplay=""
          class="video-element"
          id="camera-video"
          playsinline=""
        />,
        "tag": 5,
        "treeBaseDuration": 0,
        "type": "video",
        "updateQueue": null,
      },
      "index": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": Object {
        "Component": [Function],
        "context": null,
        "props": Object {
          "cameraOpts": Object {
            "alpha": 1,
            "camera": "rear",
            "height": 1365.3333333333333,
            "previewDrag": false,
            "tapPhoto": true,
            "toBack": true,
            "width": 1024,
            "x": 0,
            "y": 0,
          },
          "setCameraReady": [MockFunction],
        },
        "refProp": null,
        "wrappingComponentProps": null,
      },
      "memoizedState": Object {
        "context": null,
        "mount": true,
        "props": Object {
          "cameraOpts": Object {
            "alpha": 1,
            "camera": "rear",
            "height": 1365.3333333333333,
            "previewDrag": false,
            "tapPhoto": true,
            "toBack": true,
            "width": 1024,
            "x": 0,
            "y": 0,
          },
          "setCameraReady": [MockFunction],
        },
        "wrappingComponentProps": null,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "Component": [Function],
        "context": null,
        "props": Object {
          "cameraOpts": Object {
            "alpha": 1,
            "camera": "rear",
            "height": 1365.3333333333333,
            "previewDrag": false,
            "tapPhoto": true,
            "toBack": true,
            "width": 1024,
            "x": 0,
            "y": 0,
          },
          "setCameraReady": [MockFunction],
        },
        "refProp": null,
        "wrappingComponentProps": null,
      },
      "ref": null,
      "return": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 125,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 125,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": null,
          "_debugSource": null,
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": [Circular],
          "child": null,
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 0,
          "elementType": null,
          "expirationTime": 1073741823,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": null,
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": null,
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": FiberRootNode {
            "callbackExpirationTime": 0,
            "callbackNode": null,
            "callbackPriority": 90,
            "containerInfo": <div>
              <video
                autoplay=""
                class="video-element"
                id="camera-video"
                playsinline=""
              />
            </div>,
            "context": Object {},
            "current": [Circular],
            "finishedExpirationTime": 0,
            "finishedWork": null,
            "firstPendingTime": 0,
            "firstSuspendedTime": 0,
            "hydrate": false,
            "interactionThreadID": 17,
            "lastExpiredTime": 0,
            "lastPingedTime": 0,
            "lastSuspendedTime": 0,
            "memoizedInteractions": Set {},
            "nextKnownPendingLevel": 0,
            "pendingChildren": null,
            "pendingContext": null,
            "pendingInteractionMap": Map {},
            "pingCache": null,
            "tag": 0,
            "timeoutHandle": -1,
          },
          "tag": 3,
          "treeBaseDuration": 0,
          "type": null,
          "updateQueue": Object {
            "baseQueue": Object {
              "callback": null,
              "expirationTime": 1073741823,
              "next": [Circular],
              "payload": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "cameraOpts": Object {
                        "alpha": 1,
                        "camera": "rear",
                        "height": 1365.3333333333333,
                        "previewDrag": false,
                        "tapPhoto": true,
                        "toBack": true,
                        "width": 1024,
                        "x": 0,
                        "y": 0,
                      },
                      "setCameraReady": [MockFunction],
                    }
                  }
                  refProp={null}
                  wrappingComponentProps={null}
                />,
              },
              "priority": 97,
              "suspenseConfig": null,
              "tag": 0,
            },
            "baseState": null,
            "effects": null,
            "shared": Object {
              "pending": null,
            },
          },
        },
        "child": [Circular],
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 0,
        "firstEffect": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 132,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": [Circular],
          "_debugSource": Object {
            "columnNumber": 4,
            "fileName": "/Users/fabianvolkers/Developer/face-the-facts/mobile-frontend/face-the-facts/src/components/CameraView/CameraFeed.tsx",
            "lineNumber": 143,
          },
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": null,
          "child": null,
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 128,
          "elementType": "video",
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "autoPlay": true,
            "className": "video-element",
            "id": "camera-video",
            "playsInline": true,
          },
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "autoPlay": true,
            "className": "video-element",
            "id": "camera-video",
            "playsInline": true,
          },
          "ref": Object {
            "current": <video
              autoplay=""
              class="video-element"
              id="camera-video"
              playsinline=""
            />,
          },
          "return": [Circular],
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": <video
            autoplay=""
            class="video-element"
            id="camera-video"
            playsinline=""
          />,
          "tag": 5,
          "treeBaseDuration": 0,
          "type": "video",
          "updateQueue": null,
        },
        "index": 0,
        "key": null,
        "lastEffect": [Circular],
        "memoizedProps": null,
        "memoizedState": Object {
          "element": <WrapperComponent
            Component={[Function]}
            context={null}
            props={
              Object {
                "cameraOpts": Object {
                  "alpha": 1,
                  "camera": "rear",
                  "height": 1365.3333333333333,
                  "previewDrag": false,
                  "tapPhoto": true,
                  "toBack": true,
                  "width": 1024,
                  "x": 0,
                  "y": 0,
                },
                "setCameraReady": [MockFunction],
              }
            }
            refProp={null}
            wrappingComponentProps={null}
          />,
        },
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": FiberRootNode {
          "callbackExpirationTime": 0,
          "callbackNode": null,
          "callbackPriority": 90,
          "containerInfo": <div>
            <video
              autoplay=""
              class="video-element"
              id="camera-video"
              playsinline=""
            />
          </div>,
          "context": Object {},
          "current": [Circular],
          "finishedExpirationTime": 0,
          "finishedWork": null,
          "firstPendingTime": 0,
          "firstSuspendedTime": 0,
          "hydrate": false,
          "interactionThreadID": 17,
          "lastExpiredTime": 0,
          "lastPingedTime": 0,
          "lastSuspendedTime": 0,
          "memoizedInteractions": Set {},
          "nextKnownPendingLevel": 0,
          "pendingChildren": null,
          "pendingContext": null,
          "pendingInteractionMap": Map {},
          "pingCache": null,
          "tag": 0,
          "timeoutHandle": -1,
        },
        "tag": 3,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": Object {
          "baseQueue": null,
          "baseState": Object {
            "element": <WrapperComponent
              Component={[Function]}
              context={null}
              props={
                Object {
                  "cameraOpts": Object {
                    "alpha": 1,
                    "camera": "rear",
                    "height": 1365.3333333333333,
                    "previewDrag": false,
                    "tapPhoto": true,
                    "toBack": true,
                    "width": 1024,
                    "x": 0,
                    "y": 0,
                  },
                  "setCameraReady": [MockFunction],
                }
              }
              refProp={null}
              wrappingComponentProps={null}
            />,
          },
          "effects": null,
          "shared": Object {
            "pending": null,
          },
        },
      },
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": WrapperComponent {
        "_reactInternalFiber": [Circular],
        "_reactInternalInstance": Object {},
        "context": Object {},
        "props": Object {
          "Component": [Function],
          "context": null,
          "props": Object {
            "cameraOpts": Object {
              "alpha": 1,
              "camera": "rear",
              "height": 1365.3333333333333,
              "previewDrag": false,
              "tapPhoto": true,
              "toBack": true,
              "width": 1024,
              "x": 0,
              "y": 0,
            },
            "setCameraReady": [MockFunction],
          },
          "refProp": null,
          "wrappingComponentProps": null,
        },
        "refs": Object {},
        "state": Object {
          "context": null,
          "mount": true,
          "props": Object {
            "cameraOpts": Object {
              "alpha": 1,
              "camera": "rear",
              "height": 1365.3333333333333,
              "previewDrag": false,
              "tapPhoto": true,
              "toBack": true,
              "width": 1024,
              "x": 0,
              "y": 0,
            },
            "setCameraReady": [MockFunction],
          },
          "wrappingComponentProps": null,
        },
        "updater": Object {
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
        },
      },
      "tag": 1,
      "treeBaseDuration": 0,
      "type": [Function],
      "updateQueue": Object {
        "baseQueue": null,
        "baseState": Object {
          "context": null,
          "mount": true,
          "props": Object {
            "cameraOpts": Object {
              "alpha": 1,
              "camera": "rear",
              "height": 1365.3333333333333,
              "previewDrag": false,
              "tapPhoto": true,
              "toBack": true,
              "width": 1024,
              "x": 0,
              "y": 0,
            },
            "setCameraReady": [MockFunction],
          },
          "wrappingComponentProps": null,
        },
        "effects": null,
        "shared": Object {
          "pending": null,
        },
      },
    },
    "selfBaseDuration": 0,
    "sibling": null,
    "stateNode": [Circular],
    "tag": 1,
    "treeBaseDuration": 0,
    "type": [Function],
    "updateQueue": Object {
      "baseQueue": null,
      "baseState": null,
      "effects": null,
      "shared": Object {
        "pending": null,
      },
    },
  },
  "_reactInternalInstance": Object {},
  "context": Object {},
  "props": Object {
    "cameraOpts": Object {
      "alpha": 1,
      "camera": "rear",
      "height": 1365.3333333333333,
      "previewDrag": false,
      "tapPhoto": true,
      "toBack": true,
      "width": 1024,
      "x": 0,
      "y": 0,
    },
    "setCameraReady": [MockFunction],
  },
  "ref": Object {
    "current": <video
      autoplay=""
      class="video-element"
      id="camera-video"
      playsinline=""
    />,
  },
  "refs": Object {},
  "state": null,
  "stream": undefined,
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
